static/src/js/components/Map/Map.jsx

```diff
@@ -286,6 +330,59 @@ const Map = ({
   const mapRef = useRef()
   const mapElRef = useRef()
 
+  useEffect(() => {
+    // Set initial base layer visibility
+    worldImageryLayer.setVisible(base.worldImagery)
+    correctedReflectanceLayer.setVisible(base.trueColor)
+    landWaterMapLayer.setVisible(base.landWaterMap)
+
+    // Set initial overlay layer visibility
+    bordersRoadsLayer.setVisible(overlays.referenceFeatures)
+    coastlinesLayer.setVisible(overlays.coastlines)
+    placeLabelsLayer.setVisible(overlays.referenceLabels)
+  }, [])
+
+  const handleLayerChange = ({ id, checked }) => {
+    // Handle base layers
+    if (baseLayerIds.includes(id)) {
+      const newBase = {
+        worldImagery: false,
+        trueColor: false,
+        landWaterMap: false
+      }
+
+      // Set the selected base layer to true in the Redux state
+      if (id === mapLayers.worldImagery) newBase.worldImagery = checked
```

Comment:
You can do this without so many conditionals, `newBase[id] = checked` does the same thing as these 3 conditional checks